src/util/desktophelper.cpp

```diff
@@ -0,0 +1,169 @@
+        }
+#else
+        return ""; // no special command use QDesktopServices";
+#endif
+    }
+
+    QString removeChildDir(const QString& path) {
+        int index = path.lastIndexOf(QChar('/'));
+        return path.left(index);
+    }
+
+#ifdef Q_OS_LINUX
+    bool selectInFreedesktop(const QString& path) {
+        const QUrl fileurl = QUrl::fromLocalFile(path);
+        const QStringList args(fileurl.toString());
+        QDBusMessage msg = QDBusMessage::createMethodCall(
+                "org.freedesktop.FileManager1",
+                "/org/freedesktop/FileManager1",
+                "org.freedesktop.FileManager1",
+                "ShowItems");
+        msg << args << "";
+        const QDBusMessage response = QDBusConnection::sessionBus().call(msg);
+        const bool success = (response.type() != QDBusMessage::MessageType::ErrorMessage);
+        return success;
+    }
+
+    bool selectInXfce(const QString& path) {
+        const QString folder = QFileInfo(path).dir().absolutePath();
+        const QString filename = QFileInfo(path).fileName();
+
+        QDBusMessage msg = QDBusMessage::createMethodCall(
+                "org.xfce.FileManager",
+                "/org/xfce/FileManager",
+                "org.xfce.FileManager",
+                "DisplayFolderAndSelect");
+        msg << folder << filename << "" << "";
+        qDebug() << "Calling:" << msg;
+        const QDBusMessage response = QDBusConnection::sessionBus().call(msg);
+        const bool success = (response.type() != QDBusMessage::MessageType::ErrorMessage);
+        return success;
+    }
+#endif
+
+} // anonymous namespace
+
+
+namespace mixxx {
+
+void DesktopHelper::openInFileBrowser(const QStringList& paths) {
+    if (sSelectInFileBrowserCommand.isNull()) {
+        sSelectInFileBrowserCommand = getSelectInFileBrowserCommand();
+    }
+    QSet<QString> openedDirs;
+    QString dirPath;
+    for (const auto& path: paths) {
+        dirPath = removeChildDir(path);
+        // First try to select the file in file browser
+
+#ifdef Q_OS_LINUX
+        if (sSelectInFileBrowserCommand == kSelectInFreedesktop &&
+                QFile::exists(path)) {
+            if (!openedDirs.contains(dirPath)) {
+                if (selectInFreedesktop(path)) {
+                    openedDirs.insert(dirPath);
+                    break;
+                } else {
+                    qDebug() << "Select File via Freedesktop DBus interface failed";
+                }
+            }
+        }
+        if (sSelectInFileBrowserCommand == kSelectInXfce &&
+                QFile::exists(path)) {
+            if (!openedDirs.contains(dirPath)) {
+                if (selectInXfce(path)) {
+                    openedDirs.insert(dirPath);
+                    break;
+                } else {
+                    qDebug() << "Select file via Xfce DBus interface failed";
+                }
+            }
+        }
+#endif
+        if (sSelectInFileBrowserCommand.length() > 2 &&
```

Comment:
Is the >2 just preventing the kSelectInXfce or kSelectInFreedesktop case? If so, wouldn't they be handled above and so we know we can't see it here?

Reply:
Yes, I will fix it.